image-rs/src/signature/mechanism/simple/mod.rs

```diff
@@ -10,6 +10,17 @@ use serde::*;
 use strum_macros::Display;
 use strum_macros::EnumString;
 
+#[cfg(feature = "signature-simple")]
+use base64::Engine;
+
+#[cfg(feature = "signature-simple-xrss")]
+use oci_distribution::Reference;
+#[cfg(feature = "signature-simple-xrss")]
+use reqwest::Client;
+
+#[cfg(feature = "signature-simple-xrss")]
+use crate::signature::image::digest::Digest;
+
```

Comment:
There are many `#[cfg(feature = "signature-simple-xrss")]` sentences in this file. If these feature-specific sentences can be put into a separate file, it can be better for reading and maintaining. We only need to mark the `mod xrss;` sentence with `#[cfg(feature = "signature-simple-xrss")]`.

Reply:
Hi @Xynnn007 I moved most of the code in `simple/mod.rs` that was behind the `signature-simple-xrss` feature to a new file, `xrss.rs`. I marked 2 sentences with `#[cfg(feature = "signature-simple-xrss")]` in addition to the `mod xrss;` sentence in `mod.rs`:

If the XRSS feature is used then `crate::resource::get_resource(&config.sigstore_config).await` can return an error, otherwise we cannot allow it to return an error, so I left `if sig_store_config_file.is_err() { return Ok(()); }` marked with `#[cfg(feature = "signature-simple-xrss")]`.

If XRSS is not used then we don't want to create an `xrss::RegistryClient` and try to get signatures from the registry. So I created a code block where the  `xrss::RegistryClient` is created and used to try and get signatures from the registry, and I marked that code block with `#[cfg(feature = "signature-simple-xrss")]`.